### PR TITLE
[int-set] Extend the IntSet fuzzer to exercise remaining public api methods.

### PR DESCRIPTION
**Pull request by garretrieger**
Further extends the IntSet fuzzer to allow it to exercise the remaining parts of the public API that the current implementation does not cover:
- Inverted sets (invert(), is_inverted())
- Limited domain types (including discontinous domain types).
- Inclusive Iteration.

Additionally address some comments from the review of #1008 including switching to use "Cursor".

### CODE REVIEW

File: fuzz/fuzz_targets/fuzz_int_set.rs [hunk 1]
```diff
@@ -525,124 +832,175 @@ impl Operation for IntersectOp {
     }
 }
 
-/* ### End of Ops ### */
+/* ### Invert ### */
 
-fn read_u8(data: &[u8]) -> (Option<u8>, &[u8]) {
-    if data.is_empty() {
-        return (None, data);
+struct InvertOp();
```
Comment (cmyr) [inline fuzz/fuzz_targets/fuzz_int_set.rs:837]: totally academic but you can declare a 'unit struct' that has no fields with the syntax, `struct InvertOp;`.

File: int-set/src/lib.rs [hunk 2]
```diff
@@ -152,19 +163,29 @@ impl<T: Domain<T>> IntSet<T> {
     /// Returns an iterator over all disjoint ranges of values within the set in sorted ascending order.
     pub fn iter_ranges(&self) -> impl Iterator<Item = RangeInclusive<T>> + '_ {
         let u32_iter = match &self.0 {
-            Membership::Inclusive(s) => RangeIter::Inclusive {
-                ranges: s.iter_ranges(),
-            },
+            Membership::Inclusive(s) => {
```
Comment (cmyr) [inline int-set/src/lib.rs:166]: what's the relationship between these changes and the nominal goals of the patch?

**Review by cmyr — APPROVED** [review]
Looks good, my only question is whether or not this really wants to be two PRs, or whether the  `RangeIter` changes and the fuzzing changes are directly related in some way?

**Comment by garretrieger** [comment]
> Looks good, my only question is whether or not this really wants to be two PRs, or whether the `RangeIter` changes and the fuzzing changes are directly related in some way?

The RangeIter changes fixes a few bugs that the additional fuzzer functional exposed. If it's helpful I can split it off into it's own PR.

File: int-set/src/lib.rs (hunk 2 above)
Comment (garretrieger) [reply int-set/src/lib.rs:166]: This is a fix for incorrect behaviour found by the fuzzer additions. In this case the iteration over ranges discontinous was incorrect: ranges that were actually adjacent (but not in u32 space) were being returned as disjoint.

**Comment by garretrieger** [comment]
> > Looks good, my only question is whether or not this really wants to be two PRs, or whether the `RangeIter` changes and the fuzzing changes are directly related in some way?
> 
> The RangeIter changes fixes a few bugs that the additional fuzzer functional exposed. If it's helpful I can split it off into it's own PR.

Moved the IntSet changes to #1014

**Pull request merged**